frontend/src/scenes/data-warehouse/new/sourceWizardLogic.tsx

```diff
@@ -932,16 +985,39 @@ export const sourceWizardLogic = kea<sourceWizardLogicType>([
                 })
                 actions.setStep(2)
             }
+            if (searchParams.kind == 'salesforce' && searchParams.code) {
+                actions.selectConnector(SOURCE_DETAILS['Salesforce'])
+                actions.handleRedirect(searchParams.kind, {
+                    code: searchParams.code,
+                    subdomain: searchParams.subdomain,
+                })
+                actions.setStep(2)
+            }
         },
     })),
     forms(({ actions, values }) => ({
         sourceConnectionDetails: {
             defaults: buildKeaFormDefaultFromSourceDetails(SOURCE_DETAILS),
             errors: (sourceValues) => {
+                if (
+                    values.selectedConnector?.name === 'Salesforce' &&
+                    values.source.payload.code &&
+                    values.source.payload.subdomain
+                ) {
+                    return {}
+                }
```

Comment:
I understand why we have this, but 🤮 - we may wanna think about how to make oauth-based sources nicer during onboarding - likely to happen when we onboard more sources

Reply:
made it a config also